src/webextension/survey/components/fields/reason.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { formValueSelector } from 'redux-form';
+
+export const REASONS = [
+  { min: 4, value: 'fast', name: 'It is fast' },
+  { max: 4, value: 'slow', name: 'It is slow' },
+  { min: 4, value: 'works', name: 'It works well' },
+  { max: 4, value: 'broken', name: 'It is broken' },
+  { min: 4, value: 'like', name: 'I like it' },
+  { max: 4, value: 'dislike', name: 'I do not like it' },
+  { value: 'other', name: 'Other' }
+];
+
+class Reason extends Component {
+  shouldRender() {
+    const { max, min, sentiment } = this.props;
+    return min && sentiment >= min ||
+      max && sentiment <= max ||
+      (!sentiment || !max && !min);
+  }
+
+  render() {
+    if (this.shouldRender()) {
+      const { name, value } = this.props;
+      return <option value={value}>{name}</option>;
+    }
+    return null;
+  }
+}
+
+Reason.propTypes = {
+  max: React.PropTypes.number,
+  min: React.PropTypes.number,
+  name: React.PropTypes.string.isRequired,
+  sentiment: React.PropTypes.number,
+  value: React.PropTypes.string.isRequired
+};
+
+class ReasonField extends Component {
```

Comment:
Eh, it's not exported, so I'd say they're "internal" classes for the exported public component. I'd use a rule of thumb that there should be 1 public component per file.